# Eliminacja wartości odstających

In [56]:
load train.txt
tlab = train(:, 2:end);
means = mean(tlab)

for i=1:size(tlab,2)
sprintf("min %d for class %d is at %d", min(tlab(:, i)), i, find(tlab(:, i)==min(tlab(:, i))))
end

for i=1:size(tlab,2)
sprintf("max %d for class %d is at %d", max(tlab(:, i)), i, find(tlab(:, i)==max(tlab(:, i))))
end

load test.txt
telab = test(:, 2:end);
means = mean(telab)

for i=1:size(telab,2)
sprintf("min %d for class %d is at %d", min(telab(:, i)), i, find(telab(:, i)==min(telab(:, i))))
end

for i=1:size(telab,2)
sprintf("max %d for class %d is at %d", max(telab(:, i)), i, find(telab(:, i)==max(telab(:, i))))
end

means =

 Columns 1 through 6:

   1.8679e-01   1.4839e-02   2.1045e-01   2.0882e-01   7.9658e+01   1.0604e+00

 Column 7:

   9.0846e-03

ans = min 0.125 for class 1 is at 642
ans = min 0 for class 2 is at 642
ans = min 0 for class 3 is at 642
ans = min 0 for class 4 is at 642
ans = min -5.4538e-09 for class 5 is at 393
ans = min -1.214e-07 for class 6 is at 25
ans = min -3.26424e-09 for class 7 is at 539
ans = max 5.12127 for class 1 is at 186
ans = max 25.8229 for class 2 is at 186
ans = max 382.139 for class 3 is at 186
ans = max 380.858 for class 4 is at 186
ans = max 145297 for class 5 is at 186
ans = max 1934.12 for class 6 is at 186
ans = max 16.5702 for class 7 is at 186
means =

 Columns 1 through 6:

   1.8404e-01   6.9109e-04   9.4984e-04   1.5044e-05  -7.5376e-10   2.9695e-07

 Column 7:

  -9.6408e-11

ans = min 0.148148 for class 1 is at 186min 642 for class 
ans = min 3.51422e-09 for class 2 is at 704
ans = min 7.53088e-11 for class 3 is at 816
ans = min 5.13337e-11 for

Usuwam ze zbioru uczącego próbki 642 (wartości skrajnie małe dla większości klas) oraz 186 (wartości skrajnie duże dla większości klas)